### DISCUSSION TITLE
Gradle plugin tries to use velocity templates from maven plugin (?)

### DISCUSSION THREAD

**Discussion by adamzrk** [graphql-java-generator/graphql-maven-plugin-project#187] [Q&A]
[ERROR] [org.apache.velocity.loader] ResourceManager: unable to find resource 'templates/object_type.vm.java' in any resource loader.
Such error I am given when running gradle plugin to generate client code for my graphql schema. Why gradle plugin tries to search for non-existenting files (they exist in maven-plugin). How to make it work?

**Comment by etienne-sf** [comment]
Hello,

Can you precise the plug-in configuration ?

**Reply by etienne-sf** [reply]
> Why gradle plugin tries to search for non-existenting files ?

This file is an internal file. It is used to generate some of the generate classes. 

This file is used I each build since 4 years. You're the first to report this error. 

So your build must have something special.

**Comment by adamzrk** [comment]
Hi
Thank you for reply.
The problem was related to my gradle script - there were 2 transitive
dependencies to 2 different velocity versions. After fixing that the
problem is gone. But dependencies to spring boot and sfl4j messes up still.
Is it possible to generate useable code wihout spring classes?

Best regarda

pt., 28 kwi 2023, 18:21 użytkownik Etienne ***@***.***>
napisał:

**Comment by etienne-sf** [comment]
Hello,

No there is no way to get rif of spring in the generated code. And this is not the "direction" of the plugin, as the 2.0 version is based on spring-graphql.

Can you be more specific about "dependencies to spring boot and sfl4j messes up stil" ?

Did you take a look at the Gradle tutorials ?
https://github.com/graphql-java-generator/graphql-maven-plugin-project/wiki#the-tutorials

Etienne
